changelog/copy-auto-ref.dd

```diff
@@ -0,0 +1,32 @@
+`std.algorithm.mutation.copy` Has Been Modified For `put`-style Output Ranges
+
+With this release, when the target output range in a call to 
+$(REF copy, std, algorithm, mutation) is any `put` defining output range, the
+target is taken as `auto ref`. This new overload is also `void` returning
+vs the other overloads which return the unfilled portion of the given range
+or array.
```

Comment:
This isn't exact enough. It's really if the output range is not an input range with assignable elements. By deduction, this means it defines `put`, which is a necessary condition, but not a sufficient condition to trigger the auto-refness.

Reply:
e.g. one could define an input range with assignable elements that ALSO defines `put` for some optimization. In this case, it won't be auto-ref.

Reply:
Gah! If someone defines an output range which defines `put` and also has assignable elements, logically one would assume that any library function would prioritize `put` because that's supposedly the fundamental operation of output ranges. But `copy` prioritizes the `r.front` assignment, which according to the principal of least surprise is wrong. But "fixing" it would be a breaking change! 

Thankfully, an output range which is also an input range with assignable elements is really rare.

Reply:
Well, I would expect that if it does define assignable elements, either interface will properly work in this scenario (copy by value). So I think it's OK.

The real problem would be if you somehow had an assignable element input range of `T`, but an output range of `U` ;) But I won't worry about that case...